ducktape/cluster/remoteaccount.py

```diff
@@ -26,6 +27,21 @@
 from ducktape.utils.http_utils import HttpMixin
 from ducktape.utils.util import wait_until
 from ducktape.errors import DucktapeError
+import socket
+
+
+def check_ssh(method):
+    def wrapper(self, *args, **kwargs):
+        try:
+            return method(self, *args, **kwargs)
+        except (SSHException, NoValidConnectionsError, socket.error) as e:
+            if self._custom_ssh_exception_checks:
+                self._log(logging.DEBUG, "starting ssh checks:")
+                self._log(logging.DEBUG, "\n".join(repr(f) for f in self._custom_ssh_exception_checks))
+                for func in self._custom_ssh_exception_checks:
+                    func(e, self)
+            raise
```

Comment:
should it be `raise e`? I don't remember if simple `raise` would preserve the original traceback

Reply:
i originally did `raise e` however when getting the traceback from pytest, it would show me the call stack from inside here, while with just raise, it gives the original stack trace, though i havn't tried this in python3

Reply:
https://stackoverflow.com/questions/36153805/difference-between-raise-and-raise-e

Reply:
As long as it works fine in py3 (which is what newer ducktape versions are all targeting anyway), I'm fine with it.